RealmSwift/Object.swift

```diff
@@ -0,0 +1,295 @@
+        return nil
+    }
+}
+
+// MARK: Equatable
+
+/// Returns whether both objects are equal.
+public func == <T: Object>(lhs: T, rhs: T) -> Bool {
+    return RLMObjectBaseAreEqual(lhs, rhs)
+}
+
+/// Object interface which allows untyped getters and setters for Objects.
+public final class DynamicObject : Object {
+    private var listProperties = [String: List<DynamicObject>]()
+
+    /// Returns or sets the value of the property with the given name.
+    public override subscript(key: String) -> AnyObject? {
+        get {
+            if let prop = RLMObjectBaseObjectSchema(self)[key] {
+                if prop.type == .Array {
+                    // get it or set it
+                    if let list = listProperties[key] {
+                        return list
+                    }
+                    let list = List<DynamicObject>()
+                    listProperties[key] = list
+                    return list
+
+                }
+            }
+            return RLMObjectBaseObjectForKeyedSubscript(self, key)
+        }
+        set(value) {
+            if let prop = RLMObjectBaseObjectSchema(self)[key] {
+                if prop.type == .Array {
+                    throwRealmException("Setting List properties is unsupported. Instead you can add or remove objects from the current List.")
+                }
+            }
+            RLMObjectBaseSetObjectForKeyedSubscript(self, key, value)
+        }
+    }
+}
+
+
+// Internal class. Do not use directly.
+public class ObjectUtil : NSObject {
+    @objc private class func primaryKeyForClass(type: AnyClass) -> NSString? {
+        if let type = type as? Object.Type {
+            return type.primaryKey()
+        }
+        return nil
+    }
+    @objc private class func ignoredPropertiesForClass(type: AnyClass) -> NSArray? {
+        if let type = type as? Object.Type {
+            return type.ignoredProperties() as NSArray?
+        }
+        return nil
+    }
+    @objc private class func indexedPropertiesForClass(type: AnyClass) -> NSArray? {
+        if let type = type as? Object.Type {
+            return type.indexedProperties() as NSArray?
+        }
+        return nil
+    }
+
+    // Get the names of all properties in the object which are of type List<>
+    @objc private class func getGenericListPropertyNames(obj: AnyObject) -> NSArray {
+        let reflection = reflect(obj)
+
+        var properties = [String]()
+
+        // Skip the first property (super):
+        // super is an implicit property on Swift objects
+        for i in 1..<reflection.count {
+            let mirror = reflection[i].1
+            if mirror.valueType is RLMListBase.Type {
+                properties.append(reflection[i].0)
+            }
+        }
+
+        return properties
+    }
+
+    @objc private class func initializeListProperty(object: RLMObjectBase?, property: RLMProperty?, array: RLMArray?) {
+        let list = (object as Object)[property!.name]! as RLMListBase
```

Comment:
This was discussed in the previous pr. This will be of type object and it is a coding error in the binding which should crash otherwise.